cypress/integration/email/mention_email_notification_spec.js

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+/* eslint max-nested-callbacks: ["error", 4] */
+
+import users from '../../fixtures/users.json';
+
+const cypressConfig = require('../../../cypress.json');
+
+const user1 = users['user-1'];
+const user2 = users['user-2'];
+
+const text = `@${user2.username}`;
+
+describe('Email notification', () => {
+    it('post a message that mentions a user', () => {
+        cy.apiLogin('user-1');
+        cy.visit('/');
+        cy.postMessage(`${text}{enter}`);
+    });
+
+    it('sends email notification with correct email contents', () => {
+        cy.visit(`${cypressConfig.localEmailUrl}`);
+        cy.url().should('include', `${cypressConfig.localEmailUrl}`);
+
+        verifyEmailNotification('eligendi', 'Town Square', user2, user1, text);
+    });
+});
+
+function verifyEmailNotification(teamName, channelDisplayName, mentionedUser, byUser, message, feedbackEmail = 'test@example.com') {
```

Comment:
Inbucket provides an API we can access to get messages for users (https://github.com/inbucket/inbucket/wiki/REST-GET-mailbox). I think it'd be better to leverage that than to depend on the inbucket UI.

Reply:
The responses end up looking something like this, by the way. Which should give us everything we need to verify.

```json
[
{
"mailbox": "user-1",
"id": "20190520T163605-3657",
"from": "<test@example.com>",
"to": [
"<user-1@sample.mattermost.com>"
],
"subject": "[Mattermost] Notification in Stub team on May 20, 2019",
"date": "2019-05-20T16:36:05.9791057Z",
"size": 5080
}
]
```

Reply:
Oh nice, I didn't explore that.  I'll update this PR... looks like the limitation I initially mentioned is not a blocker.